migration.md

```diff
@@ -0,0 +1,107 @@
+A guide for converting from Swift-JWT Version 2.* to Swift-JWT Version 3.
+
+## Changed Structs
+
+The following Structs have been changed:
+
+### Header:
+
+To initialise the `Header` class, instead of providing a `[HeaderKey: Any]` dictionary:
+
+```swift
+let header = Header([.typ:"JWT", .kid:"KeyID"])
+```
+
+Set the fields directly:
+```swift
+let header = Header(typ: "JWT", kid: "KeyID)
+```
+
+Instead of accessing the fields using subscripting:
+```swift
+let keyID = header["kid"]
+```
+Access the Header fields directly:
+```swift
+let keyID = header.kid
+```
+
+### Claims:
+
+`Claims` is now a protocol instead of a struct. Instead of initialising the claims struct using a `[String:Any]` dictionary:
```

Comment:
I might word these before-and-after examples like so:
In Swift-JWT 2.0, you initialize using a dictionary:
`<example>`
In Swift-JWT 3.0, instead you define an object conforming to claims:
`<example>`
Then the document reads more like a side-by-side comparison of the 2.0 and 3.0 APIs.